drivers/network/tcpip/ip/lwip_glue/sys_arch.c

```diff
@@ -32,14 +37,27 @@ u32_t sys_now(void)
 void
 sys_arch_protect(sys_prot_t *lev)
 {
+#ifdef CONFIG_SMP
```

Comment:
I believe that `CONFIG_SMP` should only be used inside the kernel and HAL, so the proper fix is probably to always assume that `tcpip.sys` will be run on an SMP system (like other kernel-mode drivers do) or to detect the number of CPU cores and call the `sys_arch_protect` using function pointers.

Reply:
Yeah, sounds reasonable.